include/internal/libspdm_macro_check.h

```diff
@@ -41,4 +41,37 @@
 #undef LIBSPDM_ASYM_ALGO_SUPPORT
 #undef LIBSPDM_DHE_ALGO_SUPPORT
 
+/*ensure at least one FIPS approved algorithm is supported*/
```

Comment:
These checks should only be performed if `LIBSPDM_FIPS_MODE` is enabled.

Reply:
Yes. That make sense. Because the device may only support sm2. Thanks.